### PR TITLE
show validation loss during training 

### PR DESCRIPTION
**Pull request by Feserich**
Hey, 
I enhanced the code, so it can show the validation loss additional to the training loss. 
This is optional and will be only displayed, if a test dataset is given to the fit() or fit_transform() method as argument (beside the training dataset). 
I think the validation loss is another important metric to evaluate the model. 
Please tell me if there are some enhancements to my code. 

Regards
Feserich

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (rajarsheem) [inline .idea/vcs.xml]: Can you please remove this file?

File: test.py [hunk 2]
```diff
@@ -12,5 +12,5 @@
 
 model = StackedAutoEncoder(dims=[200, 200], activations=['linear', 'linear'], epoch=[
                            3000, 3000], loss='rmse', lr=0.007, batch_size=100, print_step=200)
-model.fit(train_X)
```
Comment (rajarsheem) [inline test.py:15]: Maybe rename test_X to valid_X ? Can you please keep both the usages (w/ and w/o valid_x) ?

File: .idea/vcs.xml (hunk 1 above)
Comment (Feserich) [reply .idea/vcs.xml]: I removed that file in my following commit. The pull request contains 4 commits. 
In my latest commit I also changed the naming from eval_x to val_x.

File: test.py (hunk 2 above)
Comment (Feserich) [reply test.py:15]: Yes that's a good idea. 
Maybe also add an example for fit_transform()?

**Comment by Feserich** [comment]
@rajarsheem I updated test.py with some examples. If you want a different naming or comment description feel free to edit it (you, as maintainer, can also edit my pull request).  
I also fixed a bug in my latest commit. 

Do you think it is better to only put noise on the training data and not on the validation data? Because after the model is trained and you want to use it,  you also don't put extra noise to your input data. So maybe the validation metrics without noise is more meaningful. What is your opinion?

**Comment by Feserich** [comment]
@rajarsheem Do you know how to fix the Travis CI errors?
